gradle/test.gradle

```diff
@@ -35,88 +35,91 @@ gradle.taskGraph.whenReady({ graph ->
 
 
 subprojects {
-  dependencies {
-    testCompile('com.github.stefanbirkner:system-rules:' + project.'system-rules.version') {
-      exclude module: 'junit-dep'
-    }
-    testCompile 'com.google.code.tempus-fugit:tempus-fugit:' + project.'tempus-fugit.version'
-    testCompile 'org.awaitility:awaitility:' + project.'awaitility.version'
-    testCompile 'edu.umd.cs.mtc:multithreadedtc:' + project.'multithreadedtc.version'
-    testCompile 'junit:junit:' + project.'junit.version'
-    testCompile 'org.assertj:assertj-core:' + project.'assertj-core.version'
-    testCompile 'org.hamcrest:hamcrest-all:' + project.'hamcrest-all.version'
-
-    testCompile 'org.mockito:mockito-core:' + project.'mockito-core.version'
-    testCompile 'org.powermock:powermock-core:' + project.'powermock.version'
-    testCompile 'org.powermock:powermock-module-junit4:' + project.'powermock.version'
-    testCompile 'org.powermock:powermock-api-mockito2:' + project.'powermock.version'
-
-    testCompile 'pl.pragmatists:JUnitParams:' + project.'JUnitParams.version'
-
-    testRuntime('cglib:cglib:' + project.'cglib.version') {
-      exclude module: 'org.apache.ant'
-    }
-  }
-
-  test {
-    doFirst {
-      TestPropertiesWriter.writeTestProperties(buildDir, name)
-    }
-    if (project.hasProperty('testMaxParallelForks')) {
-      maxParallelForks = Integer.parseUnsignedInt(project.testMaxParallelForks)
-    } else {
-      maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1
-    }
-  }
-
   apply plugin: 'nebula.facet'
   facets {
+    // We declare each test type to have parent source "test" so that test all test configurations
+    // (gradle parallelization thread count, core classpath, etc) are inherited.
+    
+    // TODO: During review, please comment on whether, for instance, we might consider
+    // performanceTest to be a child of distributedTest, or if a flat structure is preferable.
+    // Otherwise, each will implicitly inherit "main" 
     integrationTest {
       testTaskName = 'integrationTest'
       includeInCheckLifecycle = false
+      parentSourceSet = 'test'
```

Comment:
NO!!!! REVERT!

This causes all sorts of bad in IJ and Eclipse and falls outside the standard model for test sources. XTest sources should not depend on the outputs of other xTest sources.

Reply:
Ahh.  Good to know.  I was thrown off by our apparent desire to have that structure as explicitly called out in `geode-core`'s build.

So would you say that this block should be considered and anti-pattern, and this ticket should be updated to address this issue instead?

```
// geode-core/build.gradle:217-223
dependencies {
  // Integration Tests
  integrationTestCompile sourceSets.test.output

  // Distributed Tests
  distributedTestCompile sourceSets.integrationTest.output
}
```

Reply:
Yes this is a mistake and should have been cleaned up months ago when we cleaned up all the other sub projects. Whoops!